src/bp-media/bp-media-functions.php

```diff
@@ -4062,3 +4064,70 @@ function bb_check_valid_giphy_api_key( $api_key = '', $message = false ) {
 	}
 	return (bool) ( ! is_wp_error( $cache[ $api_key ] ) && isset( $cache[ $api_key ]['response']['code'] ) && 200 === $cache[ $api_key ]['response']['code'] );
 }
+
+/**
+ * Run migration for media and video description from post table to media table.
+ *
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @param int $paged Page number.
+ *
+ * @return void
+ */
+function bb_media_description_migration( $paged = 1 ) {
+	global $wpdb, $bp, $bb_background_updater;
+
+	if ( empty( $paged ) ) {
+		$paged = 1;
+	}
+
+	$per_page = (int) apply_filters( 'bb_update_migrate_media_description', 30 );
+	$offset   = ( ( $paged - 1 ) * $per_page );
+
+	$medias = $wpdb->get_results(
+		$wpdb->prepare(
+			"SELECT id, attachment_id FROM {$bp->media->table_name} ORDER BY id ASC LIMIT %d offset %d",
+			$per_page,
+			$offset
+		)
+	);
+
+	if ( ! empty( $medias ) ) {
+		$bb_background_updater->push_to_queue(
+			array(
+				'type'     => 'migration',
+				'group'    => 'bb_update_migrate_media_description',
+				'priority' => 4,
+				'callback' => 'bb_update_migrate_media_description',
+				'args'     => array( $medias, $paged ),
+			)
+		);
+		$bb_background_updater->save()->schedule_event();
+	}
```

Comment:
run a single query instead of background jobs
`UPDATE wp_bp_media
JOIN wp_posts ON wp_posts.ID = wp_bp_media.attachment_id
SET wp_bp_media.description = wp_posts.post_content
ORDER BY wp_posts.ID DESC;`